qiita_plugins/target_gene/tgp/split_libraries/split_libraries_fastq.py

```diff
@@ -0,0 +1,269 @@
+
+    if errors:
+        raise ValueError("You have run_prefix values with multiple "
+                         "samples: %s" % ' -- '.join(errors))
+
+    return samples
+
+
+def generate_per_sample_fastq_command(forward_seqs, reverse_seqs, barcode_fps,
+                                      mapping_file, output_dir, params_str):
+    """Generates the per-sample FASTQ split_libraries_fastq.py command
+
+    Parameters
+    ----------
+    forward_seqs : list of str
+        The list of forward seqs filepaths
+    reverse_seqs : list of str
+        The list of reverse seqs filepaths
+    barcode_fps : list of str
+        The list of barcode filepaths
+    mapping_file : str
+        The path to the mapping file
+    output_dir : str
+        The path to the split libraries output directory
+    params_str : str
+        The string containing the parameters to pass to
+        split_libraries_fastq.py
+
+    Returns
+    -------
+    str
+        The CLI to execute
+
+    Raises
+    ------
+    ValueError
+        - If barcode_fps is not an empty list
+        - If there are run prefixes in the mapping file that do not match
+        the sample names
+    """
+    if barcode_fps:
+        raise ValueError('per_sample_FASTQ can not have barcodes: %s'
+                         % (', '.join(basename(b) for b in barcode_fps)))
+    sn_by_rp = get_sample_names_by_run_prefix(mapping_file)
+    samples = []
+    for f in forward_seqs:
+        f = basename(f)
+        if re.match("^[0-9]+\_.*", f):
+            # getting just the main filename
+            f = basename(f).split('_', 1)[1]
+        # removing extentions: fastq or fastq.gz
+        if 'fastq' in f.lower().rsplit('.', 2):
+            f = f[:f.lower().rindex('.fastq')]
+        # this try/except block is simply to retrieve all possible errors
+        # and display them in the next if block
+        try:
+            samples.append(sn_by_rp[f])
+            del sn_by_rp[f]
+        except KeyError:
+            pass
+
+    if sn_by_rp:
+        raise ValueError(
+            'Some run_prefix values do not match your sample names: %s'
+            % ', '.join(sn_by_rp.keys()))
+
+    cmd = str("split_libraries_fastq.py --store_demultiplexed_fastq "
+              "-i %s --sample_ids %s -o %s %s"
+              % (','.join(forward_seqs), ','.join(samples),
+                 output_dir, params_str))
+    return cmd
+
+
+def generate_split_libraries_fastq_cmd(filepaths, mapping_file, atype,
+                                       out_dir, parameters):
+    """Generates the split_libraries_fastq.py command
+
+    Parameters
+    ----------
+    filepaths : list of (str, str)
+        The artifact filepaths and their type
+    mapping_file : str
+        The artifact QIIME-compliant mapping file
+    atype : str
+        The artifact type
+    out_dir : str
+        The job output directory
+
+    Returns
+    -------
+    str
+        The CLI to execute
+
+    Raises
```

Comment:
IMOO it will be worth fixing these (document Raises) as it will not take that much time and fix others as we find them.